cfme/tests/containers/test_ssl_providers.py

```diff
@@ -49,9 +48,60 @@
 )
 
 
+@pytest.fixture(scope="module")
+def sync_ssl_certificate(provider, appliance):
+    """
+    Args:
+        provider (OpenshiftProvider): Openshift provider object,
+                                      this would be the source of the certificate.
+        appliance (IPAppliacne): CFME appliance object.
+    Returns:
+        None
+    """
+
+    # creating a ssh connection to both appliance and provider
+    provider_ip = provider.data["ipaddress"]
+    provider_cred_name = provider.data["credentials"]
+    connections_info = {"username": "root",
+                        "password": provider.get_credentials_from_config(provider_cred_name).secret,
+                        "hostname": provider_ip}
+
+    provider_ssh = ssh.SSHClient(**connections_info)
+    appliance_ssh = appliance.ssh_client()
+
+    # Connection to the applince in case of dead connection
+    if not appliance_ssh.connected:
+        appliance_ssh.connect()
```

Comment:
Is it possible that this connection could be dead as well?

Reply:
It's probably not connected by default?

Reply:
@jawatts @gshefer yes, this possible to get a not-connected ssh object

Reply:
This is a demo of getting an SSH client from an appliance

``` python
In [1]: import cfme.utils.appliance as appl
   ...: appl.CREATE_IS_PEDANTIC = False
   ...: app = appl.get_or_create_current_appliance()
   ...: app_ssh = app.ssh_client()
   ...: print app_ssh.connected
   ...: 
[WARNING] cfme/utils/appliance/__init__.py:2762: DeprecationWarning: Using class 'CurrentAppliance' (either directly or via inheritance) is deprecated: The CurrentAppliance descriptor is being phased outin favour of collections.
  appliance = CurrentAppliance()
 (/home/snaim/virtEns/myDevEnv/lib/python2.7/site-packages/debtcollector/_utils.py:61)
[WARNING] cfme/utils/appliance/__init__.py:2712: ApplianceSummoningWarning: magical creation of appliance objects has been deprecated, please obtain a appliance object directly
  category=ApplianceSummoningWarning,
 (cfme/utils/appliance/__init__.py:2712)
[WARNING] cfme/utils/appliance/__init__.py:2700: DeprecationWarning: Your appliance specification has old-style base_url, please change
  return load_appliances(appliances, global_kwargs)
 (cfme/utils/appliance/__init__.py:2619)
[WARNING] /home/snaim/virtEns/myDevEnv/lib/python2.7/site-packages/yaycl_crypt/__init__.py:148: YayclCryptWarning: yaml "./conf/credentials.yaml" and eyaml present for "./conf/credentials.eyaml" config. Ignoring encrypted yaml.
  warnings.warn(warn_msg, YayclCryptWarning)
 (/home/snaim/virtEns/myDevEnv/lib/python2.7/site-packages/yaycl_crypt/__init__.py:148)
[WARNING] /home/snaim/virtEns/myDevEnv/lib/python2.7/site-packages/oslo_i18n/_message.py:21: ImportWarning: Not importing directory '/home/snaim/virtEns/myDevEnv/lib/python2.7/site-packages/oslo_i18n/locale': missing __init__.py
  import locale
 (/home/snaim/virtEns/myDevEnv/lib/python2.7/site-packages/oslo_i18n/_message.py:21)
[WARNING] /usr/lib64/python2.7/pkgutil.py:186: ImportWarning: Not importing directory './data': missing __init__.py
  file, filename, etc = imp.find_module(subname, path)
 (/usr/lib64/python2.7/pkgutil.py:186)
None

In [2]: app_ssh.connected

In [3]: print  app_ssh.connected
None

In [4]: app_ssh.connect()

In [5]: print  app_ssh.connected
True

In [6]: 
```

As you can see the SSH connection is not connected by defualt